Distribution/Server/Features/Html.hs

```diff
@@ -1474,37 +1521,70 @@ mkHtmlTags HtmlUtilities{..}
         tagList <- queryGetTagList
         let withCounts = filter ((>0) . snd) . map (\(tg, pkgs) -> (tg, Set.size pkgs)) $ tagList
             countSort = sortBy (flip compare `on` snd) withCounts
-        return $ toResponse $ Resource.XHtml $ hackagePage "Hackage tags" $
+        return $ toResponse $ Resource.XHtml $ hackagePage "Hackage tags"
           [ h2 << "Hackage tags"
           , h4 << "By name"
-          , paragraph ! [theclass "toc"] << (intersperse (toHtml ", ") $ map (tagItem . fst) withCounts)
+          , paragraph ! [theclass "toc"] << intersperse (toHtml ", ") (map (tagItem . fst) withCounts)
           , h4 << "By frequency"
-          , paragraph ! [theclass "toc"] << (intersperse (toHtml ", ") $ map (toHtml . tagCountItem) countSort)
+          , paragraph ! [theclass "toc"] << intersperse (toHtml ", ") (map (toHtml . tagCountItem) countSort)
           ]
       where tagCountItem (tg, count) =
               [ tagItem tg
               , toHtml $ " (" ++ show count ++ ")"
               ]
             tagItem tg = anchor ! [href $ tagUri tags "" tg] << display tg
 
+    putAliasEdit :: DynamicPath -> ServerPartE Response
+    putAliasEdit dpath = do
```

Comment:
All these new GETs need appropriate `cacheControl` info. In some cases because these are expensive (like the revdep graph) but the other reason is if we have no cach-control then the CDN uses some default that is not always appropriate (ie people get confused by stale data). So we have to think about it in every case one way or another.